src/main/pages/che-6/user-guide/commands-ide-macro.adoc

```diff
@@ -8,201 +8,223 @@ redirect_from: commands-ide-macro.html
+When you use a button for the first time and if this button has multiple commands defined for a goal, you are asked to choose the default command associated with this button. The consecutive clicks of the button trigger the previously selected command.
 
-The same mechanisms apply to the `Debug` button.
+To select a command to execute from a goal, click and hold the button. This command becomes the default command associated with the button.
 
-*Command Controller*
+==== Using the command controller
 
-The command controller allow you to see the state of the workspace and the last command executed. You can see since how long the command started and also decide if it should be stopped or relaunched.
+Use the command controller to view the state of the workspace and the last command that was executed. You can see the time span since a command was started and decide if it should be stopped or relaunched.
 
-When multiple commands have been executed it’s possible to see the list of all previously executed commands by clicking on the widget.
+To view the list of all the previously executed commands when multiple commands are executed, click on the widget.
 
 image::commands/command-toolbar-expanded.png[]
 
-To clean the list, remove the command’s process from the list of processes.
+To clean the list, remove the process of the command from the list of processes.
 
 image::commands/command-clean-toolbar.png[]
 
-*Preview Button*
+==== Using the Preview button
 
-If you have a command which start servers (for example, Tomcat) you can define the preview URL to access the running server. Learn more at link:servers.html#preview-urls[server preview URLs].
+For commands used to start servers, define the preview URL to access the running server. To learn more, see the link:servers.html#preview-urls[Preview URLs] section.
 
-The preview button provides quick access to all the servers that are running in workspace’s machines.
+The *Preview* button provides quick access to all the servers that are running in the machines of the workspace.
 
 [id="authoring-command-instructions"]
-== Authoring Command Instructions
+== Authoring command instructions
 
 A command may contain a single instruction or a succession of commands. For example:
 
 ----
-# each command starts from a new line
-cd /projects/spring
+cd /projects/spring				<1>
 mvn clean install
+----
+<1> Each command starts on a new line
 
-# a succession of several commands where `;` stands for a new line
-cd /projects/spring; mvn clean install
-
-# a succession of several commands where execution of a subsequent command depends on execution of a preceeding one - if there's no /projects/spring directory, `mvn clean install` won't be executed
-cd /projects/spring && mvn clean install
 ----
+cd /projects/spring; mvn clean install		<1>
+----
+<1> A succession of several commands where `;` stands for a new line
```

Comment:
Does this stand for a new line or does it simply separate commands regardless of their exit status? In Bash, these aren't quite the same thing.

Reply:
It is the equivalent of a new line, for practical purposes. I think it's more accurate to say "where separate commands are separated by ';'"